### PR TITLE
change default group

### LINKED ISSUE #436: Playbooks should use the "generic" instead of the "all" group
Since I am in the process of adding the SONiC switches of the SCS Hardware Landscape to the configuration repository, I noticed that contrary to the documentation, the "custom" group is not the authoritative one for all nodes used by OSISM.

(see ["cookiecutter" defaults](https://github.com/osism/cfg-cookiecutter/blob/main/%7B%7Bcookiecutter.project_name%7D%7D/inventory/20-roles#L9) - "The "all" group is not used in OSISM. Therefore it is important that all nodes are explicitly listed here.")

I tried to define two separate groups "network-switches" and "dhcp-static-nodes" and discovered that some roles/playbooks are trying to access/configure these nodes  

In the "ansible-playbooks" repository there are some playbooks that explicitly use the "all" group as a default group.
```
$ git grep "'all'" |head -4
playbooks/generic/auditd.yml:    - "{{ hosts_auditd|default('all') }}"
playbooks/generic/bootstrap.yml:  hosts: "{{ hosts_bootstrap | default('all') }}"
playbooks/generic/bootstrap.yml:        gather_subset: "{{ osism_setup_gather_subset | default('all') }}"
playbooks/generic/bootstrap.yml:        gather_subset: "{{ osism_setup_gather_subset | default('all') }}"

$ git grep "hosts: all" |head -4
library/osquery.py:- hosts: all
playbooks/generic/auditd.yml:  hosts: all
playbooks/generic/certificates.yml:  hosts: all
playbooks/generic/check-reboot.yml:  hosts: all
...

With a specific group default of "generic" on all playbooks the "generic" groups works as described in the default and it is possible to keep the old behavior or to have a dedicated group.

### PR DESCRIPTION
**Pull request by scoopex**
close #436 

This pull request addresses #436.

I replaced the "all" group by `default(hosts_default_group|default('generic'))` with a regex replace.
I simply adjusted the defaults in the following way:
```
$ find . -type f -name "*.yml" -exec sed -i "~s,hosts: all,hosts: \"{{ default(hosts_default_group|default('generic')) }}\",g" {} \;
$ find . -type f -name "*.yml" -exec sed -i "~s,|default('all'),|default(hosts_default_group|default('generic')),g" {} \;
```
As I haven't been working on this for very long, it may well take a little more work, but perhaps we can dare to do so :-)

### CODE REVIEW

**Comment by scoopex** [comment]
Sorry, i accidentally missed to push a rework.
Now it should be as intended.

@berendt ready for review

**Comment by scoopex** [comment]
Damn, it seems that a chained "default" is not accepted by "ansible-lint" but valid.

**Review by berendt — CHANGES_REQUESTED** [review]

File: playbooks/generic/facts.yml [hunk 1]
```diff
@@ -10,7 +10,7 @@
 
 - name: Gather facts for all hosts
   ignore_unreachable: true
-  hosts: "{{ hosts_facts|default('all') }}"
+  hosts: "{{ hosts_facts|default(hosts_default_group|default('generic')) }}"
```
[the remark this answers is not in the document]
Comment (berendt) [reply playbooks/generic/facts.yml:13]: We can also overwrite this with osism/defaults + hosts_facts parameter.

File: playbooks/generic/dump-facts.yml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 ---
 - name: Print all available facts
   ignore_unreachable: true
-  hosts: "{{ hosts_facts|default('all') }}"
+  hosts: "{{ hosts_facts|default(hosts_default_group|default('generic')) }}"
```
[the remark this answers is not in the document]
Comment (berendt) [reply playbooks/generic/dump-facts.yml:4]: We can also overwrite this with osism/defaults + hosts_facts parameter.

**Pull request merged**